frontend/src/main.ts

```diff
@@ -3,24 +3,59 @@ import { createApp } from 'vue';
 import awsExports from './aws-exports';
 
 import App from '@/App.vue';
-import router from '@/router';
+import { historyRouter, hashRouter } from '@/router';  // Import the two routers
 
 import 'bootstrap';
 
 import PrimeVue from 'primevue/config';
 import ConfirmationService from 'primevue/confirmationservice';
 import ToastService from 'primevue/toastservice';
-
-import 'primevue/resources/primevue.min.css';
-import 'primevue/resources/themes/bootstrap4-light-blue/theme.css';
-import './assets/styles/styles.scss';
+import { VueQueryPlugin, QueryClient } from '@tanstack/vue-query'
+import { isAxiosError } from 'axios';
+import { THREE_HOURS } from '@/constants/TimeUnits';
 
 Amplify.configure(awsExports); // Config Amplify for Cognito resource.
 
+// Create the QueryClient instance with default options
+const queryClient = new QueryClient({
+    defaultOptions: {
+        queries: {
+            refetchOnMount: false,
+            refetchOnWindowFocus: false,
+            retry: (failureCount, error) => {
+                const MAX_RETRIES = 3;
+                const HTTP_STATUS_TO_NOT_RETRY = [400, 401, 403, 404, 500];
```

Comment:
Retrying on a 401 response isn’t necessary because it indicates that the requester does not have valid authorization. Retrying won’t change the fact that the current token is invalid. Instead, the client should handle token refresh logic separately if authorization is expected to be restored.

Reply:
Ok, so this "retry" wasn't what I thought about retry previously. I see, I think I got it.